geocoder/geocoder.cpp

```diff
@@ -1,19 +1,38 @@
 #include "geocoder/geocoder.hpp"
 
+#include "search/house_numbers_matcher.hpp"
+
 #include "indexer/search_string_utils.hpp"
 
 #include "base/assert.hpp"
 #include "base/logging.hpp"
 #include "base/scope_guard.hpp"
+#include "base/stl_helpers.hpp"
+#include "base/string_utils.hpp"
 #include "base/timer.hpp"
 
 #include <algorithm>
+#include <set>
 #include <utility>
 
 using namespace std;
 
 namespace
 {
+size_t const kMaxResults = 100;
+
+map<geocoder::Type, double> const kWeight = {
+    {geocoder::Type::Country, 10.0},
```

Comment:
Why don't you want to use direct address hashing?

Reply:
Good point, will fix.